www/__tests__/enketoHelper.test.ts

```diff
@@ -0,0 +1,182 @@
+    const badForm = { getDataStr: () => { return '<aDxjD5f5KAghquhAvsormy xmlns:jr="http://openrosa.org/javarosa" xmlns:orx="http://openrosa.org/xforms" id="time_use_survey_form_v9_1"><start>2023-10-13T15:05:48.890-06:00</start><end>2023-10-13T15:05:48.892-06:00</end><group_hg4zz25><Start_date>2016-08-25</Start_date><Start_time>17:24:32.928-06:00</Start_time><End_date>2016-07-25</End_date><End_time>17:30:31.000-06:00</End_time><Activity_Type>personal_care_activities</Activity_Type><Personal_Care_activities>doing_sport</Personal_Care_activities><Employment_related_a_Education_activities/><Domestic_activities/><Recreation_and_leisure/><Voluntary_work_and_care_activities/><Other/></group_hg4zz25><meta><instanceID>uuid:dc16c287-08b2-4435-95aa-e4d7838b4225</instanceID><deprecatedID/></meta></aDxjD5f5KAghquhAvsormy>'}};
+    const config = {
+        survey_info: {
+          surveys: {
+            TimeUseSurvey: { compatibleWith: 1, 
+              formPath: "https://raw.githubusercontent.com/sebastianbarry/nrel-openpath-deploy-configs/surveys-info-and-surveys-data/survey-resources/data-json/time-use-survey-form-v9.json", 
+              labelTemplate: {en: "{ erea, plural, =0 {} other {# Employment/Education, } }{ da, plural, =0 {} other {# Domestic, } }",
+                              es: "{ erea, plural, =0 {} other {# Empleo/Educación, } }{ da, plural, =0 {} other {# Actividades domesticas, }}"}, 
+              labelVars: {da: {key: "Domestic_activities", type: "length"},
+                          erea: {key: "Employment_related_a_Education_activities", type:"length"}}, 
+              version: 9}
+          }
+        }
+      };
+    const opts = { timelineEntry: { end_local_dt: {timezone: "America/Los_Angeles"}, start_ts: 1469492672.928242, end_ts: 1469493031}};
+    
+    console.log(config);
+    expect(saveResponse(surveyName, form, config, opts)).resolves.toMatchObject({label: "1 Personal Care", name: "TimeUseSurvey"});
+    expect(saveResponse(surveyName, badForm, config, opts)).resolves.toMatchObject({message: "The times you entered are invalid. Please ensure that the start time is before the end time."});
+});
+
+/*
+* We retrieve all the records every time instead of caching because of the
+* usage pattern. We assume that the demographic survey is edited fairly
+* rarely, so loading it every time will likely do a bunch of unnecessary work.
+* Loading it on demand seems like the way to go. If we choose to experiment
+* with incremental updates, we may want to revisit this.
+*/
+//   export function loadPreviousResponseForSurvey(dataKey: string) {
+it('loads the previous response to a given survey', () => {
+    //not really sure if I can test this yet given that it relies on an angular service...
+    // loadPreviousResponseForSurvey("manual/demographic_survey");
```

Comment:
Do you want to just wait for https://github.com/e-mission/e-mission-phone/pull/1064 ?

Reply:
yes, I think that will allow me to make some progress testing this function